### DISCUSSION TITLE
Quickemu Availability ON Windows?

### DISCUSSION THREAD

**Discussion by p134c0d3** [quickemu-project/quickemu#1267] [Ideas]
# Expected behavior

To be able to use QuickEMU on my Windows installation QEMU

## Actual behavior

Unable to find where to install on Windows.

**Comment by lj3954** [comment]
Quickemu may be used under WSL2. Native windows support is not planned, especially considering QEMU on the platform only supports software virtualization.

**Comment by jacklul** [comment]
whpx accelerator works fine on windows with Hypervisor Platform optional feature installed.

After few modifications to the script (mainly removing unix-only options) the script launches alpine VM fine.

But it will probably be a lot of work hours to achieve full compatibility with all supported options, probably not worth the effort.

**Comment by lj3954** [comment]
> whpx accelerator works fine on windows with Hypervisor Platform optional feature installed.
> 
> After few modifications to the script (mainly removing unix-only options) the script launches alpine VM fine.
> 
> But it will probably be a lot of work hours to achieve full compatibility with all supported options, probably not worth the effort.

Please provide some documentation on that. 

The bash version of quickemu of course wouldn't be of any use on Windows regardless, but I've been working on a rewrite of Quickemu in Rust (https://github.com/lj3954/quickemu-rs) and adding Windows support certainly isn't out of the question if QEMU does indeed support hardware virtualization on the platform.

**Reply by jacklul** [reply]
> Please provide some documentation on that.

https://www.qemu.org/docs/master/system/introduction.html
WSL2 uses the same hypervisor, so the performance should be comparable.

There was also support for Intel's HAXM but it was removed in 8.2.0 - https://www.qemu.org/docs/master/about/removed-features.html#haxm-accel-hax-removed-in-8-2

> I've been working on a rewrite of Quickemu in Rust (https://github.com/lj3954/quickemu-rs)

Nice!
